docs/src/components/Network.js

```diff
@@ -64,40 +67,178 @@ const getProtocolFee = async (provider, unlockAddress) => {
+        await wallets[0].getEthereumProvider()
+      )
+      const signer = await provider.getSigner()
+
+      // switch network
+      // await wallets[0].switchChain(network.id)
+      await provider.send('wallet_switchEthereumChain', [
+        {
+          chainId: `0x${network.id.toString(16)}`,
+        },
+      ])
+
+      // Send tx
+      const unlock = new Contract(
+        network.unlockAddress,
+        [
+          {
+            inputs: [
+              { internalType: 'address', name: 'token', type: 'address' },
+              { internalType: 'uint256', name: 'amount', type: 'uint256' },
+              { internalType: 'uint24', name: 'poolFee', type: 'uint24' },
+            ],
+            name: 'swapAndBurn',
+            outputs: [],
+            stateMutability: 'nonpayable',
+            type: 'function',
+          },
+          {
+            inputs: [],
+            name: 'swapBurnerAddress',
+            outputs: [{ internalType: 'address', name: '', type: 'address' }],
+            stateMutability: 'view',
+            type: 'function',
+          },
+        ],
+        signer
+      )
+
+      const addressOfTokenToBurn = token.address || ethers.ZeroAddress
+      let tx
+
+      if (addressOfTokenToBurn === ethers.ZeroAddress) {
+        try {
+          tx = await unlock.swapAndBurn.populateTransaction(
+            token.address || ethers.ZeroAddress,
+            balance,
+            3000 // No need to enter a pool fee for native currency
+          )
+        } catch (e) {
+          console.error(e)
+        }
+      } else {
+        const fees = [500, 3000, 10000]
+        for (let fee of fees) {
+          if (!tx) {
+            try {
+              tx = await unlock.swapAndBurn.populateTransaction(
+                token.address || ethers.ZeroAddress,
+                balance,
+                fee // pool for the token, not for UDT/UP
+              )
+            } catch (e) {
+              console.error(e)
+            }
+          }
+        }
+      }
+      if (tx) {
+        try {
+          const transaction = await signer.sendTransaction(tx)
+          setHash(transaction.hash)
+          await transaction.wait()
+        } catch (error) {
+          console.error(error)
+          // TODO: replace with Toast once we have UI package
+          alert('The transaction to burn tokens could not be sent.')
+        }
+        await reload()
+        setBurning(false)
+      } else {
+        // TODO: replace with Toast once we have UI package
+        alert('We could find a way to burn this token.')
+      }
+      setBurning(false)
     }
+  }
+
+  useEffect(() => {
+    if (burning && authenticated) {
+      // resume burning!
+      burn()
+    }
+  }, [burning, authenticated])
+
+  return (
+    <li>
+      {Number(formatUnits(balance, token.decimals)).toFixed(4)} {token.symbol}{' '}
+      <Button isLoading={true} onClick={burn}>
+        Burn
+      </Button>{' '}
```

Comment:
here could add a condition to show the button only if the swap burner is set correctly in Unlock